deps/rabbitmq_mqtt/test/shared_SUITE.erl

```diff
@@ -446,12 +447,11 @@ events(Config) ->
     QueueNameBin = <<"mqtt-subscription-", ClientId/binary, "qos0">>,
     QueueName = {resource, <<"/">>, queue, QueueNameBin},
     [E2, E3 | E4] = get_events(Server),
-    QueueType = case rabbit_ct_helpers:is_mixed_versions(Config) of
-                    false ->
+    QueueType = case is_feature_flag_enabled(Config, rabbit_mqtt_qos0_queue) of
```

Comment:
But if you do this, you never test that the feature flag can be enabled in a cluster, even though all of the nodes support it.

Reply:
Why not?

When this test runs in non-mixed version mode, all nodes in the cluster support this feature flag and this feature flag gets enabled by default without me explicitly enabling it within the test suite.
This feature flag does not contain any migration code. But even if it contained migration code, and that migration code failed, this whole cluster would not become healthy and the test case will fail (which is what we want).

In mixed version mode, there can be 3 situations:
1. Not all nodes support this feature flag (as it's the case with the currently configured 3.11.7 "lower version" nodes): In this case, I do want this test to assert that a `classic` queue gets created.
2. All nodes support this feature flag (this will be the case in the future when we bump the "lower version" nodes to 3.12). Even in that scenario I do want the feature flag to be disabled to test that a classic queue gets created (feature flags are disabled by default in our mixed version cluster tests).
3. The feature flag becomes `required` (even further in the future), in which case the feature flag is enabled, and the test will pass as well. Once we `require` this feature flag, this case statement will be removed as part of "cleaning up" the feature flag's compatibility code.

Reply:
There is a 4th situation: all nodes support this feature flag but don't run the same code, in which case we would also want to make sure they can work together with the feature flag enabled.

Perhaps this situation is not possible today; do you confirm that?

Reply:
>There is a 4th situation: all nodes support this feature flag but don't run the same code

This corresponds to situation 2. in my comment above, right?
In that case, what you're suggesting is that instead of only testing with the feature flag being disabled, we could (or should) **additionally** test with the feature flag being enabled?

We could do that. However this sounds to me that we are missing this scenario for **all test suites**.
So, instead of testing what we have today:
- all nodes on the new version with all feature flags enabled, and
-  mixed version nodes with all feature flags disabled by default

we need yet another GitHub action workflow with
- mixed version nodes with all supported feature flags enabled

We could introduce such a new workflow if we think it's worth it?

From my point of view, what you are suggesting is good, but it applies to all feature flags in all test suites, and not only to this single feature flag in this single test case we're discussing, right?

Reply:
> This corresponds to situation 2. in my comment above, right?

Yes, this is close.

> From my point of view, what you are suggesting is good, but it applies to all feature flags in all test suites, and not only to this single feature flag in this single test case we're discussing, right?

I think it would more be efficient to handle that in each testsuite. There is no need to run testcases which are not affected by feature flags again. Also some testcases may only be relevant in a specific situation, not all. We execute all testsuites twice for regular and mixed-version cluster because the conditions are very different even when considering the compilation.